tests/test_butterfly_network.py

```diff
@@ -0,0 +1,26 @@
+import asyncio
+from pytest import mark
+from math import log
+import random
+
+
+@mark.asyncio
+async def test_butterfly_network(sharedatadir):
+    import honeybadgermpc.apps.shuffle.butterfly_network as butterfly
+    from honeybadgermpc.passive import generate_test_triples, Field, runProgramAsTasks
+
+    async def verify_output(ctx, **kwargs):
```

Comment:
This test scenario looks great. It uses some preshared inputs, preshared -1/1 shares, run the butterfly network, and check that the output is indeed a permutation (the sorted arrays match).